Modules/getpath.c

```diff
@@ -1296,16 +1296,28 @@ calculate_zip_path(PyCalculatePath *calculate)
 {
     PyStatus res;
 
+    // wsprintf() is not available on all platforms:
```

Comment:
I honestly find this code harder to follow and more complex than the preprocessor solution